src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+                            draft.getSku(), SkuKeyTuple.of(draft).getKey()), ex);
+                        return null;
+                    });
+            }
+        } else {
+            statistics.incrementFailed();
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    /**
+     * Tries to create Inventory Entry in CTP project, using {@code draft}.
+     * Before calling API, the channel reference from {@code draft} is replaced, so it points to proper channel ID
+     * in target system.
+     * It either creates inventory entry and increments "created" counter in statistics, or increments "failed" counter
+     * and executes error callback function in case of any exception.
+     *
+     * @param draft draft of new inventory entry
+     * @return {@link CompletionStage} instance that indicates method progress
+     */
+    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
+    private CompletionStage<Void> attemptCreate(final InventoryEntryDraft draft) {
+        final Optional<InventoryEntryDraft> fixedDraft = replaceChannelReference(draft);
+        if (fixedDraft.isPresent()) {
+            return inventoryService.createInventoryEntry(fixedDraft.get())
+                .thenAccept(createdEntry -> statistics.incrementCreated())
+                .exceptionally(ex -> {
+                    statistics.incrementFailed();
+                    handleFailure(format("Failed to create inventory entry of sku '%s' and supply channel key '%s'",
+                        draft.getSku(), SkuKeyTuple.of(draft).getKey()), ex);
+                    return null;
+                });
+        } else {
+            statistics.incrementFailed();
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    /**
+     * Returns {@link Optional} that may contain {@link InventoryEntryDraft}. The payload of optional would be:
+     * <ul>
+     *     <li>Same {@code draft} instance if it has no reference to supply channel</li>
+     *     <li>New {@link InventoryEntryDraft} instance if {@code draft} contains reference to supply channel.
+     *     New instance would have same values as {@code draft} except for supply channel reference. Reference
+     *     will be replaced with reference that points to ID of existing channel for key given in draft.</li>
+     *     <li>Empty if supply channel for key wasn't found in {@code supplyChannelKeyToId}</li>
+     * </ul>
+     *
+     * @param draft inventory entry draft from processed list
+     * @return {@link Optional} with draft that is prepared to being created or compared with existing
+     * {@link InventoryEntry}, or empty optional if method fails to find supply channel ID that should be referenced
+     */
+    private Optional<InventoryEntryDraft> replaceChannelReference(final InventoryEntryDraft draft) {
+        final String supplyChannelKey = SkuKeyTuple.of(draft).getKey();
```

Comment:
I will document how valid input is supposed to look like.

So basically `of(InventoryEntryDraft)` performs good. If there is supplyChannel it takes either 'key' from object from expanded reference (what isn't supposed to be null) or `Id` from not expanded reference. Otherwise it takes `null` when there is no supply channel reference at all.
In fact, there is mistake in `of(InventoryEntry)` -> when reference is not expanded there then `key` results in `null` what could points to another resource... I think that in that case the most appropriate solution is to throw exception, and fail sync of such resource.